src/core/ir/jit/model.rs

```diff
@@ -1,7 +1,80 @@
 use std::fmt::{Debug, Formatter};
+use std::hash::{Hash, Hasher};
 
+use tailcall_hasher::TailcallHasher;
+
+use super::Result;
 use crate::core::ir::model::IR;
 
+const EMPTY_VEC: &Vec<Field<Children>> = &Vec::new();
+#[derive(Clone, Debug)]
+pub struct ExecutionPlan {
+    // A representation where nodes are connect to their parent (one to one)
+    pub parent: Vec<Field<Parent>>,
+
+    // A representation where nodes are connect to their children (one to many)
+    pub children: Vec<Field<Children>>,
+
+    // A pre-resolved value of the execution plan
+    #[allow(unused)]
+    pub value: Option<Result<Vec<u8>>>,
```

Comment:
why to serialize and deserialize to store and read the value? Can't we directly keep serde_json_borrowed::Value?